pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -0,0 +1,66 @@
+package csidrivercontrollerservicecontroller
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/client-go/informers/core/v1"
+
+	opv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehash"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// WithObservedProxyHook injects into the deployment's containers the observed proxy config.
+func WithObservedProxyHook() DeploymentHookFunc {
+	return func(opSpec *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deploymentCopy, err := v1helpers.InjectObservedProxyInDeploymentContainers(
+			deployment,
+			opSpec.ObservedConfig.Raw,
+			"csidrivercontrollerservicecontroller",
+			"proxy",
+		)
+		if err != nil {
+			return err
+		}
+		*deployment = *deploymentCopy
+		return nil
+	}
+}
+
+// With SecretHashAnnotationHook creates a deployment hook that annotates a Deployment with a secret's hash.
+func WithSecretHashAnnotationHook(
+	namespace string,
+	secretName string,
+	secretInformer corev1.SecretInformer,
+) DeploymentHookFunc {
+	return func(opSpec *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferenceFromLister(
+			nil,
+			secretInformer.Lister(),
+			resourcehash.NewObjectRef().ForSecret().InNamespace(namespace).Named(secretName),
+		)
+		if err != nil {
+			return fmt.Errorf("invalid dependency reference: %w", err)
+		}
+		if deployment.Annotations == nil {
+			deployment.Annotations = map[string]string{}
+		}
+		if deployment.Spec.Template.Annotations == nil {
+			deployment.Spec.Template.Annotations = map[string]string{}
+		}
+		for k, v := range inputHashes {
+			annotationKey := fmt.Sprintf("operator.openshift.io/dep-%s", k)
```

Comment:
lets move this into a var/constant

Reply:
I'd like to leave this on as is. I prefer to leave format strings inline, plus I couldn't re-use the const for the annotation below (`%x` != `%s`)